[Xcode] Add timestamp outputs for verifier script phases

emw-apple opened the pull request:
#### c57c92851648a35916fc322f535fe3192000044a
<pre>
[Xcode] Add timestamp outputs for verifier script phases
<a href="https://bugs.webkit.org/show_bug.cgi?id=261229">https://bugs.webkit.org/show_bug.cgi?id=261229</a>
rdar://115076525

Reviewed by Alexey Proskuryakov.

This prevents them from re-running on every build, and from causing
module verification to re-run on every build.

Additionally, fix up some input dependencies to these scripts. &quot;Check
For Framework Include Consistency&quot; and &quot;Check For Inappropriate Macros
in External Headers&quot; should only re-run when headers change, so have
them depend on the all-product-headers.yaml VFS file generated by Xcode.

Where possible, move &quot;Check For Inappropriate Files In Framework&quot; phases
to run *outside* of its framework target. This script looks for
arbitrary files in the framework bundle, including those not tracked by
the build system. As a result, it&apos;s not possible to express input
dependencies, and continuing to run it as part of the framework build
causes the module verifier to re-run needlessly (XCBuild conservatively
assumes that any script phase without outputs declared may be modifying
framework headers).

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Tools/Scripts/check-for-inappropriate-macros-in-external-headers:
* Tools/Scripts/check-for-webkit-framework-include-consistency:

Canonical link: <a href="https://commits.webkit.org/267740@main">https://commits.webkit.org/267740@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/10cc70b2c13c3deac4d3ceeff7a1b587c5ba0bf5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17475 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17800 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18316 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19267 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16351 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/17672 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21078 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17944 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18487 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17682 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18002 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15176 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20087 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15235 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15898 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22542 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/15106 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16245 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16067 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20380 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/16685 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16653 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14118 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/19040 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15783 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/4425 "Failed to checkout and rebase branch from PR 17510") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4180 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20153 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/20267 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16491 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/4266 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
I assume that these timestamps won't defeat any caching technologies like ccache?

emw-apple committed:
[Xcode] Add timestamp outputs for verifier script phases
https://bugs.webkit.org/show_bug.cgi?id=261229
rdar://115076525

Reviewed by Alexey Proskuryakov.

This prevents them from re-running on every build, and from causing
module verification to re-run on every build.

Additionally, fix up some input dependencies to these scripts. "Check
For Framework Include Consistency" and "Check For Inappropriate Macros
in External Headers" should only re-run when headers change, so have
them depend on the all-product-headers.yaml VFS file generated by Xcode.

Where possible, move "Check For Inappropriate Files In Framework" phases
to run *outside* of its framework target. This script looks for
arbitrary files in the framework bundle, including those not tracked by
the build system. As a result, it's not possible to express input
dependencies, and continuing to run it as part of the framework build
causes the module verifier to re-run needlessly (XCBuild conservatively
assumes that any script phase without outputs declared may be modifying
framework headers).

* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKitLegacy/WebKitLegacy.xcodeproj/project.pbxproj:
* Tools/Scripts/check-for-inappropriate-macros-in-external-headers:
* Tools/Scripts/check-for-webkit-framework-include-consistency:

Canonical link: https://commits.webkit.org/267740@main
